app/javascript/shared/components/useMediaQuery.js

```diff
@@ -0,0 +1,50 @@
+import { useState, useEffect } from 'preact/hooks';
+
+/**
+ * Pre-defined breakpoints for width.
+ *
+ * Note: These were copied from _generator.scss.
+ */
+export const BREAKPOINTS = Object.freeze({
+  Small: '640',
+  Medium: '768',
+  Large: '1024',
+});
+
+/**
+ * A custom Preact hook for evaluating whether or not a CSS media query is matched or not.
+ *
+ * @param {string} query The media query to evaluate.
+ * @param {function} render A render prop for using   the media query.
+ *
+ * @returns {boolean} True if the media query is matched, false otherwise.
+ *
+ * @example
+ * import { useMediaQuery } from '@components/useMediaQuery';
+ *
+ * function SomeComponent({ query, render }) {
```

Comment:
```suggestion
 * function SomeComponent({ query }) {
```

Don't think we need the render prop here